doc/doxygen/src/creating-modules.md

```diff
@@ -0,0 +1,42 @@
+Creating modules                                            {#creating-modules}
+================
+The general structure                                  {#the-general-structure}
+=====================
+Like @ref creating-an-application "applications", modules are directories
+containing source files and a Makefile. Additionally their API can be defined
+in one or more header files, residing in the include path of their
+super-module.
+
+E.g. the @ref sys_shell module is implemented in `sys/shell` and defines its
+API in `sys/include/shell.h`.
```

Comment:
The only information that I was/am missing here was a hint that a module must not only be a system or kernel module but also a e.g. driver. IMO no need for a complete list here. The diver porting guide will also state that a driver might become a module